controlExtensionProject/src/test/java/controlExtension/RadioButtonPage.java

```diff
@@ -0,0 +1,46 @@
+package controlExtension;
+
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+import org.openqa.selenium.support.How;
+
+import framework.PageObjectBase;
+
+public class RadioButtonPage extends PageObjectBase{
+	private String URL = "https://demoqa.com/radio-button";
+
+	@FindBy(how = How.ID, using = "yesRadio")
+	private WebElement btnYes;
+	@FindBy(how = How.ID, using = "impressiveRadio")
+	private WebElement btnImpressive;
+	@FindBy(how = How.ID, using = "noRadio")
+	private WebElement btnNo;
+	@FindBy(how = How.XPATH, using = "//p[@class='mt-3']")
+	private WebElement btnselected;
+
+	protected RadioButtonPage(WebDriver driver) {
+		super(driver);
+	}
+
+	public RadioButtonPage navigate() {
+		this.driver.get(URL);
+		return this;
+	}
+
+	public void select(String btn) {
+		if(btn == "Yes") {
```

Comment:
This pattern violates OpenClosed principle. A better solution is to pass the RadioButton label then find the element based on that label.